api/v1alpha1/llamastack_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// LlamaStackSpec defines the desired state of LlamaStack
+type LlamaStackSpec struct {
+	Replicas     *int32       `json:"replicas,omitempty"`
+	Distribution string       `json:"distribution"` // e.g., "ollama", "vllm-gpu"
+	Image        string       `json:"image"`
+	Port         *int32       `json:"port,omitempty"`
+	Resources    ResourceSpec `json:"resources,omitempty"`
+}
+
+// LlamaStackStatus defines the observed state of LlamaStack
+type LlamaStackStatus struct {
+	Distribution string   `json:"distribution"`
+	ActiveAPIs   []string `json:"activeAPIs,omitempty"`
+	Ready        bool     `json:"ready"`
+}
+
+// ResourceSpec defines resource requests/limits
+type ResourceSpec struct {
+	Requests ResourceQuantity `json:"requests,omitempty"`
+	Limits   ResourceQuantity `json:"limits,omitempty"`
+}
+
+type ResourceQuantity struct {
+	CPU    string `json:"cpu,omitempty"`
+	Memory string `json:"memory,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// LlamaStack is the Schema for the llamastacks API
+type LlamaStack struct {
```

Comment:
```suggestion
type LlamaStackDistribution struct {
```
See my earlier comment around naming.